### PR TITLE
Validate +required and +format=dns-label declaratively

### PR DESCRIPTION
**Pull request by thockin**
This does exactly 1 new field to demonstrate.

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go [hunk 1]
```diff
@@ -71,6 +71,14 @@ func IsQualifiedName(value string) []string {
 	return errs
 }
 
+func ValidateRequired[T comparable](fldPath *field.Path, value T) field.ErrorList {
+	var zero T
+	if value == zero {
```
Comment (jpbetz) [inline staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go:76]: Great use of generics.

**Review by jpbetz — COMMENTED** [review]
The generator use looks right to me.

File: staging/src/k8s.io/code-generator/cmd/validation-gen/validators/openapi.go [hunk 2]
```diff
@@ -66,8 +67,12 @@ func (openAPIDeclarativeValidator) ExtractValidations(t *types.Type, comments []
 }
 
 func FormatValidationFunction(format string) FunctionGen {
+	//FIXME: Let's define the naming convention for these - jsonStyle or GoStyle?
```
Comment (jpbetz) [inline staging/src/k8s.io/code-generator/cmd/validation-gen/validators/openapi.go:70]: For OpenAPI compatibility of format names I believe we want `lower-case`. https://json-schema.org/draft/2020-12/json-schema-validation#name-defined-formats

Comment (thockin) [reply staging/src/k8s.io/code-generator/cmd/validation-gen/validators/openapi.go:70]: that suggests "dns-1123-label" or "dns1123-label" as a convention?

**Comment by jpbetz** [comment]
LGTM!

**Pull request merged**